validator.go

```diff
@@ -445,6 +445,27 @@ OUTER:
 			}
 
 		default:
+			switch current.Kind() {
+			case reflect.Ptr, reflect.Interface:
+				if !ct.runValidationWhenNil && current.IsNil() {
+					v.errs = append(v.errs,
+						&fieldError{
```

Comment:
Not very familiar with the errors in this library, so I just went by what was used in the other instance where the `runValidationWhenNil` is checked and could be entirely wrong 😄